Carbon.HttpClient.Auth/README.Md

```diff
@@ -0,0 +1,119 @@
+﻿# Carbon.HttpClient.Auth
+
+Simple HTTP Handler for inter-microservices communication. Recommended for only purpose of in-cluster communication.
+Supports oAuth2 client_credentials grant typed authorization. Both HTTP 1.1 (REST-based) and HTTP 2.0 (GRPC-based)
+when [Carbon.WebApplication.Grpc.Client](../Carbon.WebApplication.Grpc.Client/README.Md) used are supported.
+
+It handles the token retreival from identity server supported by oauth2 and 
+In case of having 401 status code from microservices due to token expiration or something, it refreshes the token and resends
+the http request again.
+
+[Carbon.WebApplication.Grpc.Client](../Carbon.WebApplication.Grpc.Client/README.Md) uses this package as-is for the
+inter-microservices authorized communication.
+
+**1a. Simply register this httpclient (For HTTP 1.1 REST-based)**
+ ```csharp
+//IServiceCollection services
+services.AddHttpClientAuth();
+```
+or..
+
+**1b. Or register this grpchttpclient (For HTTP 2.0 GRPC-based)**
+ ```csharp
+//IServiceCollection services
+services.AddGrpcHttpClientAuth();
+```
+
+**2. Create an authentication with the name just as it is writing in the configuration file**
+
+This creates the token at startup by communicating with identity server and stores it until you ask for.
+```csharp
+//IApplicationBuilder
+app.CreateAuthentication("RAM");
```

Comment:
How can we tell that this is an exemplary name?

Reply:
Will be introduced in the next PR